From 20e3427d655572d9f5b194278419cd2cfd8902bb Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Tue, 27 Mar 2012 13:01:53 +0800
Subject: [PATCH 28/35] powerpc/p2041rdb: add the rman and srio device node in
 usdpaa dts

This is based on the p2041rdb-usdpaa.dts in FSL DPAA SDK 1.1.

Add the rman and srio device node in usdpaa dts. These will be used by
the fra application. And also add the bman buffers used by the rman.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/boot/dts/p2041rdb-usdpaa.dts |   70 +++++++++++++++++++++++++++++
 1 file changed, 70 insertions(+)

diff --git a/arch/powerpc/boot/dts/p2041rdb-usdpaa.dts b/arch/powerpc/boot/dts/p2041rdb-usdpaa.dts
index 725c7bc..f9e5a2f 100644
--- a/arch/powerpc/boot/dts/p2041rdb-usdpaa.dts
+++ b/arch/powerpc/boot/dts/p2041rdb-usdpaa.dts
@@ -74,6 +74,7 @@
 		bman = &bman;
 		qman = &qman;
 		pme = &pme;
+		rman = &rman;
 		sdhc = &sdhc;
 		msi0 = &msi0;
 		msi1 = &msi1;
@@ -254,6 +255,21 @@
 			fsl,bpool-cfg = <0 0 0 2112 0 0xfeedabba>;
 			fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
 		};
+		bp10: buffer-pool@10 {
+			compatible = "fsl,p2041-bpool", "fsl,bpool";
+			fsl,bpid = <10>;
+			fsl,bpool-thresholds = <0x10 0x30 0x0 0x0>;
+		};
+		bp11: buffer-pool@11 {
+			compatible = "fsl,p2041-bpool", "fsl,bpool";
+			fsl,bpid = <11>;
+			fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+		};
+		bp12: buffer-pool@12 {
+			compatible = "fsl,p2041-bpool", "fsl,bpool";
+			fsl,bpid = <12>;
+			fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+		};
 	};
 
 	qman-portals@ff4200000 {
@@ -820,6 +836,37 @@
 			gpio-controller;
 		};
 
+		rman: rman@1e0000 {
+			compatible = "fsl,rman";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x1e0000 0x20000>;
+			reg = <0x1e0000 0x20000>;
+			interrupts = <16 2 1 11>; /* err_irq */
+			fsl,qman-channels-id = <0x62 0x63>;
+
+			inbound-block@0 {
+				compatible = "fsl,rman-inbound-block";
+				reg = <0x0 0x800>;
+			};
+			global-cfg@b00 {
+				compatible = "fsl,rman-global-cfg";
+				reg = <0xb00 0x500>;
+			};
+			inbound-block@1000 {
+				compatible = "fsl,rman-inbound-block";
+				reg = <0x1000 0x800>;
+			};
+			inbound-block@2000 {
+				compatible = "fsl,rman-inbound-block";
+				reg = <0x2000 0x800>;
+			};
+			inbound-block@3000 {
+				compatible = "fsl,rman-inbound-block";
+				reg = <0x3000 0x800>;
+			};
+		};
+
 		usb0: usb@210000 {
 			compatible = "fsl,p2041-usb2-mph",
 					"fsl,mpc85xx-usb2-mph", "fsl-usb2-mph";
@@ -1282,6 +1329,29 @@
 		};
 	};
 
+	rapidio@ffe0c0000 {
+		compatible = "fsl,srio";
+		interrupts = <16 2 1 11>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+		reg = <0xf 0xfe0c0000 0 0x11000>;
+
+		port1 {
+			#address-cells = <2>;
+			#size-cells = <2>;
+			cell-index = <1>;
+			ranges = <0 0 0xc 0x20000000 0 0x10000000>;
+		};
+
+		port2 {
+			#address-cells = <2>;
+			#size-cells = <2>;
+			cell-index = <2>;
+			ranges = <0 0 0xc 0x30000000 0 0x10000000>;
+		};
+	};
+
 	localbus@ffe124000 {
 		compatible = "fsl,p2041-elbc", "fsl,elbc", "simple-bus";
 		reg = <0xf 0xfe124000 0 0x1000>;
-- 
1.7.9.7

